File: moderate/counting_primes.py
#!/usr/bin/env python2
# CodeEval - Counting Primes
# https://www.codeeval.com/open_challenges/63/

from sys import argv

def is_prime(number):
    ''' Return True if number is a prime, else False'''
    if number < 2: 
        return False

    # Only need to check up to square root
    limit = int(number**0.5 + 1)

    for i in range(2, limit):
        if number%i == 0:
            return False
    return True


def main():
    with open(argv[1], "rt") as f:
        for line in f:
            _ = line.strip().split(",")
            integers = [int(i) for i in _]
            integers = range(integers[0], integers[1]+1)
            count = 0
            for integer in integers:
                if is_prime(integer):
                    count += 1
            print(count)


if __name__ == "__main__":
    main()

File: moderate/counting_primes.cs
/* CodeEval - Counting Primes
 * https://www.codeeval.com/open_challenges/63/
 */

using System.IO;
using System.Collections.Generic;

class Program
{
    static bool IsPrime(int some_int)
    {
        if (some_int < 2)
            return false;
        int limit = (int)System.Math.Sqrt(some_int) + 1;        
        for (int i = 2; i < limit; i++)
            if (some_int % i == 0)
                return false;
        return true;
    }

    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string[] numbers = line.Trim().Split(',');            
            int x = int.Parse(numbers[0]);
            int y = int.Parse(numbers[1]);
            int count = 0;
            for (int i = x; i <= y; i++)
                if (IsPrime(i))
                    count++;
            System.Console.WriteLine(count);
        }
    }
}
